readthedocs/proxito/middleware.py

```diff
@@ -73,17 +73,22 @@ def add_proxito_headers(self, request, response):
         if cache_tags:
             response['Cache-Tag'] = ','.join(cache_tags)
 
-        if hasattr(request, 'rtdheader'):
-            response['X-RTD-Project-Method'] = 'rtdheader'
-        elif hasattr(request, 'subdomain'):
-            response['X-RTD-Project-Method'] = 'subdomain'
-        elif hasattr(request, 'cname'):
-            response['X-RTD-Project-Method'] = 'cname'
-
-        if hasattr(request, 'external_domain'):
-            response['X-RTD-Version-Method'] = 'domain'
-        else:
-            response['X-RTD-Version-Method'] = 'path'
+        unresolved_domain = request.unresolved_domain
+        if unresolved_domain:
+            origin_to_description = {
+                OriginType.http_header: "rtdheader",
+                OriginType.custom_domain: "cname",
+                OriginType.public_domain: "subdomain",
+                OriginType.external_domain: "subdomain",
+            }
+            response["X-RTD-Project-Method"] = origin_to_description[
+                unresolved_domain.origin
+            ]
```

Comment:
if we don't care about keeping the old names, we could also just use `unresolved_domain.origin.name`

Reply:
I think nobody should be using this. We've used internally to know where we were getting this value from. I'm fine using new names here, but probably worth double checking with @ericholscher in case he has other concerns about this.

Reply:
Yea, this is mostly just for debugging on our side. I don't think anything should be depending on it, but `validatehttp` would be the only place I'd think.